src/Converter/ShipmentLineItemsConverter.php

```diff
@@ -50,14 +57,28 @@ private function convertUnitRefundToLineItem(ShipmentRefund $shipmentRefund): Li
         Assert::isInstanceOf($shipment, AdjustableInterface::class);
         Assert::lessThanEq($shipmentRefund->total(), $shipment->getAdjustmentsTotal());
 
+        $grossValue = $shipmentRefund->total();
+
+        $taxAdjustments = $shipment->getAdjustments(AdjustmentInterface::TAX_ADJUSTMENT);
+
+        $taxAdjustmentAmount = 0;
+        if (count($taxAdjustments) > 0) {
+            $taxAdjustmentAmount = $this->taxRateAmountProvider->provide($taxAdjustments->first());
+        }
```

Comment:
Similar to `TaxRateProvider`, you should check if there is only one tax adjustment here